src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsListener.java

```diff
@@ -2,65 +2,110 @@
 
 import java.awt.event.KeyAdapter;
 import java.awt.event.KeyEvent;
+import java.util.ArrayList;
+import java.util.Locale;
+import java.util.stream.Collectors;
 
 /**
  * respond to grabKey and display the key binding
  */
 public class KeyBindingsListener extends KeyAdapter {
 
     private final KeyBindingTable table;
+    private boolean isDeleteKey;
+    private boolean isFunctionKey;
+    private boolean isEscapeKey;
 
     public KeyBindingsListener(KeyBindingTable table) {
         this.table = table;
     }
 
     @Override
     public void keyPressed(KeyEvent evt) {
-        // first check if anything is selected if not the return
+        // first check if anything is selected if not then return
         final int selRow = table.getSelectedRow();
         boolean isAnyRowSelected = selRow >= 0;
         if (!isAnyRowSelected) {
             return;
         }
 
-        final String modifier = KeyEvent.getKeyModifiersText(evt.getModifiers());
+        boolean isDeleteKey = false;
+        boolean isFunctionKey = false;
+        boolean isEscapeKey = false;
+        final String modifier = getModifierText(evt);
 
         // VALIDATE code and modifier
         // all key bindings must have a modifier: ctrl alt etc
         if ("".equals(modifier)) {
             int kc = evt.getKeyCode();
-            boolean isFunctionKey = (kc >= KeyEvent.VK_F1) && (kc <= KeyEvent.VK_F12);
-            boolean isEscapeKey = kc == KeyEvent.VK_ESCAPE;
-            boolean isDeleteKey = kc == KeyEvent.VK_DELETE;
+            isFunctionKey = (kc >= KeyEvent.VK_F1) && (kc <= KeyEvent.VK_F12);
+            isEscapeKey = kc == KeyEvent.VK_ESCAPE;
+            isDeleteKey = kc == KeyEvent.VK_DELETE;
             if (!(isFunctionKey || isEscapeKey || isDeleteKey)) {
                 return; // need a modifier except for function, escape and delete keys
             }
         }
 
-        final String code = KeyEvent.getKeyText(evt.getKeyCode());
-        // second key cannot be a modifiers
-        if ("Tab".equals(code)
-                || "Backspace".equals(code)
-                || "Enter".equals(code)
-                || "Space".equals(code)
-                || "Ctrl".equals(code)
-                || "Shift".equals(code)
-                || "Alt".equals(code)) {
+        int code = evt.getKeyCode();
+        String newKey;
+        //skip the event triggered only by a modifier
+        if (code == KeyEvent.VK_ALT ||
+                code == KeyEvent.VK_TAB ||
+                code == KeyEvent.VK_BACK_SPACE ||
+                code == KeyEvent.VK_ENTER ||
```

Comment:
Could you comment why `ENTER` is treated as modifier? Also applies to "TAB" and "BACK_SPACE". Maybe the reason is that these keys should not be used at shortcuts at all? Maybe a new `if` statement should be introduced with an appropriate comment.